contracts/staking/test/end_to_end_tests/deployment.ts

```diff
@@ -0,0 +1,431 @@
+            [owner],
+            new BigNumber(1),
+            constants.ZERO_AMOUNT,
+        );
+
+        // Deploy Exchange.
+        exchange = await ExchangeContract.deployFrom0xArtifactAsync(
+            exchangeArtifacts.Exchange,
+            env.provider,
+            txDefaults,
+            exchangeArtifacts,
+            new BigNumber(chainId),
+        );
+
+        // Deploy ReadOnlyProxy.
+        readOnlyProxy = await ReadOnlyProxyContract.deployFrom0xArtifactAsync(
+            stakingArtifacts.ReadOnlyProxy,
+            env.provider,
+            txDefaults,
+            stakingArtifacts,
+        );
+
+        // Deploy Staking.
+        staking = await StakingContract.deployFrom0xArtifactAsync(
+            stakingArtifacts.Staking,
+            env.provider,
+            txDefaults,
+            stakingArtifacts,
+        );
+
+        // Deploy the staking proxy.
+        stakingProxy = await StakingProxyContract.deployFrom0xArtifactAsync(
+            stakingArtifacts.StakingProxy,
+            env.provider,
+            txDefaults,
+            stakingArtifacts,
+            staking.address,
+            readOnlyProxy.address,
+        );
+
+        // Deploy the asset proxy contracts.
+        erc20Proxy = await ERC20ProxyContract.deployFrom0xArtifactAsync(
+            assetProxyArtifacts.ERC20Proxy,
+            env.provider,
+            txDefaults,
+            assetProxyArtifacts,
+        );
+        erc721Proxy = await ERC721ProxyContract.deployFrom0xArtifactAsync(
+            assetProxyArtifacts.ERC721Proxy,
+            env.provider,
+            txDefaults,
+            assetProxyArtifacts,
+        );
+        erc1155Proxy = await ERC1155ProxyContract.deployFrom0xArtifactAsync(
+            assetProxyArtifacts.ERC1155Proxy,
+            env.provider,
+            txDefaults,
+            assetProxyArtifacts,
+        );
+        multiAssetProxy = await MultiAssetProxyContract.deployFrom0xArtifactAsync(
+            assetProxyArtifacts.MultiAssetProxy,
+            env.provider,
+            txDefaults,
+            assetProxyArtifacts,
+        );
+        staticCallProxy = await StaticCallProxyContract.deployFrom0xArtifactAsync(
+            assetProxyArtifacts.StaticCallProxy,
+            env.provider,
+            txDefaults,
+            assetProxyArtifacts,
+        );
+
+        // Set up the staking wrapper so that the entire staking interface can be accessed
+        // easily through the proxy.
+        stakingWrapper = new StakingContract(stakingProxy.address, env.provider);
+    });
+
+    describe('deployment and configuration', () => {
+        describe('exchange specific', () => {
+            // Registers an asset proxy in the exchange contract and ensure that the correct state changes occurred.
+            async function registerAssetProxyAndAssertSuccessAsync(
+                registrationContract: any, // TODO(jalextowle): Express this in a type-safe way (during a debt-demolition)
```

Comment:
Optional, but I think this works: 
https://gist.github.com/mzhu25/1db8353388b43eaf6a752d25059d3e9b